### PR TITLE
🛺 Migrate to nextjs so we can eventually deploy to vercel

### PR DESCRIPTION
**Pull request by Hacksore**
if we are using RSCs we can do the following
- remove tanstack shiz for querying
- create a github PAT and do data fetching on the server

home:
image

user:
image

### CODE REVIEW

**Comment by ajchili** [comment]
noooo dude where did my hooks go omg......

**Comment by Hacksore** [comment]
> noooo dude where did my hooks go omg......

Your hooks were mid

**Comment by ajchili** [comment]
i'll hook you

**Comment by Hacksore** [comment]
We are getting there!

**Comment by ajchili** [comment]
love it

**Comment by Hacksore** [comment]
image

**Comment by Hacksore** [comment]
@ajchili make sure to create a `GITHUB_PAT` and add it to the vercel env when you plan on deploying.

File: src/app/[username]/opengraph-image.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { notFound } from "next/navigation";
+// Route segment config
+export const runtime = "edge";
+
+// Image metadata
+export const alt = "GitPilled OG Image";
```
Comment (ajchili) [inline src/app/[username]/opengraph-image.tsx:6]: copy pasta yikes.

**Review by ajchili — APPROVED** [review]

**Pull request merged**